lib/sablon/processor/document.rb

```diff
@@ -175,35 +177,37 @@ def initialize(fields)
         end
 
         def operations
-          while @fields.any?
-            @operations << consume(true)
-          end
+          @operations << consume(true) while @fields.any?
           @operations.compact
         end
 
         def consume(allow_insertion)
           @field = @fields.shift
           return unless @field
+
           case @field.expression
           when /^=/
-            if allow_insertion
-              Statement::Insertion.new(Expression.parse(@field.expression[1..-1]), @field)
-            end
+            Statement::Insertion.new(Expression.parse(@field.expression[1..]), @field) if allow_insertion
           when /([^ ]+):each\(([^ ]+)\)/
-            block = consume_block("#{$1}:endEach")
-            Statement::Loop.new(Expression.parse($1), $2, block)
-          when /([^ ]+):if\(([^)]+)\)/
-            block = consume_block("#{$1}:endIf")
-            Statement::Condition.new(Expression.parse($1), block, $2)
-          when /([^ ]+):if/
-            block = consume_block("#{$1}:endIf")
-            Statement::Condition.new(Expression.parse($1), block)
+            block = consume_block("#{::Regexp.last_match(1)}:endEach")
+            Statement::Loop.new(Expression.parse(::Regexp.last_match(1)), ::Regexp.last_match(2), block)
+          when /([^ ]+):if\(([^)]+)\)$/
+            block = consume_block("#{::Regexp.last_match(1)}:endIf")
+            Statement::Condition.new(Expression.parse(::Regexp.last_match(1)), block, ::Regexp.last_match(2))
+          when /([^ ]+):if$/
+            block = consume_block("#{::Regexp.last_match(1)}:endIf")
+            Statement::Condition.new(Expression.parse(::Regexp.last_match(1)), block)
           when /comment/
-            block = consume_block("endComment")
+            block = consume_block('endComment')
             Statement::Comment.new(block)
           when /^@([^ ]+):start/
-            block = consume_block("@#{$1}:end")
-            Statement::Image.new(Expression.parse($1), block)
+            block = consume_block("@#{::Regexp.last_match(1)}:end")
+            Statement::Image.new(Expression.parse(::Regexp.last_match(1)), block)
+          when /([^ ]+):if\s*(==|!=|<=|>=|<|>)\s*(['"].+?['"]|[\w.]+)$/
+            block = consume_block("#{::Regexp.last_match(1)}:endIf")
+            Statement::ExpressiveCondition.new(
+              ::Regexp.last_match(1), ::Regexp.last_match(2), ::Regexp.last_match(3), block
+            )
```

Comment:
Match a mergefield beginning with `<context_field>:if` followed by any of the given operators, and then either a value surrounded by quotes or another context field.